app/src/main/java/com/lightricks/feedexercise/network/FeedApiService.kt

```diff
@@ -1,5 +1,34 @@
 package com.lightricks.feedexercise.network
 
-/**
- * todo: add the FeedApiService interface and the Retrofit and Moshi code here
- */
+import com.lightricks.feedexercise.data.FeedItemAdapter
+import com.squareup.moshi.Moshi
+import com.squareup.moshi.kotlin.reflect.KotlinJsonAdapterFactory
+import io.reactivex.Single
+import retrofit2.Retrofit
+import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
+import retrofit2.converter.moshi.MoshiConverterFactory
+import retrofit2.http.GET
+
+
+interface FeedApiService {
+    @GET("feed.json")
+    fun getFeed(): Single<DataTransferObjects>
+
+    companion object {
+        private const val BASE_URL = "https://assets.swishvideoapp.com/Android/demo/"
+
+        fun getFeedApiService(): FeedApiService {
```

Comment:
Would you agree that in the context of the exercise, to satisfy DI pattern it will be enough to use `Application` instance for retrieving singleton instances of things like `db`, `Moshi` etc?

Reply:
There's an even easier way. In Kotlin an `object` is a singleton. 
Here's how you can do it:

First, declare a property (Kotlin `val`) of type `FeedApiService` in your `companion object` and initialise it with `retrofit.create(FeedApiService::class.java)`. Alternatively, you can declare a named `object` (e.g. `object FeedApi`). Then calling this property will always return the same service.

Second, you must move `Moshi` and `Retrofit` initialisation outside of the function body. You can make them `private val`, and initialize at the declaration spot.

Please, Slack me it something is not clear.

Reply:
I think that this exactly what I implemented in the last commit. Correct me if I'm wrong.

Reply:
Yes, I wrote this before I saw it.